spanner/transaction.go

```diff
@@ -258,6 +266,7 @@ func (t *txReadOnly) ReadWithOptions(ctx context.Context, table string, keys Key
 		if opts.DirectedReadOptions != nil {
 			directedReadOptions = opts.DirectedReadOptions
 		}
+		orderBy = opts.OrderBy
```

Comment:
@mayurkale22 
Here the request level orderBy field in ReadOptions is directly overriding the client level orderBy field.
My concern here is, if a customer sets `client level` `orderBy` field to `NO_ORDER` and in every request if the customer passes ReadOptions with out setting `orderBy` field (which will by default have `UNSPECIFIED`) then this unset field will override the client level `NO_ORDER`.
Is this expected behavior?

Testcase:
```
{
name:   "Client level 1",
client: &ReadOptions{Index: "testIndex", Limit: 100, Priority: sppb.RequestOptions_PRIORITY_LOW, RequestTag: "testRequestTag", OrderBy: sppb.ReadRequest_ORDER_BY_NO_ORDER},
read:   &ReadOptions{Index: "testIndex", Limit: 100, Priority: sppb.RequestOptions_PRIORITY_LOW, RequestTag: "testRequestTag"},
want:   &ReadOptions{Index: "testIndex", Limit: 100, Priority: sppb.RequestOptions_PRIORITY_LOW, RequestTag: "testRequestTag", OrderBy: sppb.ReadRequest_ORDER_BY_NO_ORDER},
},
```

Here we expect the result to be `NO_ORDER`, but the actual output will be `UNSPECIFIED`.

Reply:
In short, whenever request level ReadOptions is not null, then the default value of `orderBy` field will be passed to ReadRequest.

Reply:
Thanks for catching this. Added condition to set only when it is not default (UNSPECIFIED). PTAL